### PR TITLE
community[patch],weaviate[patch]: types: the args of the `WeaviateFilter` is optional

### PR DESCRIPTION
**Pull request by hehuangz**
Fixes # (issue)

it's not necessary pass distance and where condition at the same time

### CODE REVIEW

**Comment by jacoblee93** [comment]
Thank you!

**Comment by jacoblee93** [comment]
Seems it's failing build - can you have a look?

**Comment by hehuangz** [comment]
> Seems it's failing build - can you have a look?

ok

**Comment by hehuangz** [comment]
> Seems it's failing build - can you have a look?

Hi, I don't have a better way to revise it, please check if it works, thanks!

File: libs/langchain-community/src/vectorstores/weaviate.ts [hunk 1]
```diff
@@ -88,7 +88,7 @@ interface ResultRow {
  */
 export interface WeaviateFilter {
   distance?: number;
-  where: WhereFilter;
+  where?: WhereFilter;
```
Comment (bracesproul) [inline libs/langchain-community/src/vectorstores/weaviate.ts:91]: This will be a breaking change to anyone using this and not already passing `where`

**Review by bracesproul — CHANGES_REQUESTED** [review]
Dismiss this

**Comment by jacoblee93** [comment]
Getting:
```
Error: @langchain/weaviate:build: src/vectorstores.ts(243,20): error TS2345: Argument of type '{ operands?: ...[] | undefined; operator?: "And" | "Or" | "Equal" | "Like" | "NotEqual" | "GreaterThan" | "GreaterThanEqual" | "LessThan" | "LessThanEqual" | "WithinGeoRange" | "IsNull" | "ContainsAny" | "ContainsAll" | undefined; ... 13 more ...; valueGeoRange?: { ...; } | undefined; } | undefined' is not assignable to parameter of type '{ operands?: ...[] | undefined; operator?: "And" | "Or" | "Equal" | "Like" | "NotEqual" | "GreaterThan" | "GreaterThanEqual" | "LessThan" | "LessThanEqual" | "WithinGeoRange" | "IsNull" | "ContainsAny" | "ContainsAll" | undefined; ... 13 more ...; valueGeoRange?: { ...; } | undefined; }'.
```

On CI build

**Comment by bracesproul** [comment]
Closing due to CI issues. Please open a new PR with the CI issues fixed if you'd still like to get this merged. Thank you!

**Pull request closed without merging**